crates/sui-indexer-alt/src/lib.rs

```diff
@@ -59,6 +59,10 @@ pub struct IndexerArgs {
     #[arg(long)]
     pub last_checkpoint: Option<u64>,
 
+    /// Don't write to the watermark tables for concurrent pipelines.
+    #[arg(long)]
+    pub skip_watermark: bool,
```

Comment:
This commit in particular makes me wonder what the ideal situation would be when it comes to cli args vs file-based configs, when do you put something in one vs the other, when do you put something in both (with cli arg having precedence), etc.

If you look at how `git` is configured it even allows for EnvVars for some options as well as a general purpose `--config <key>=<value>` cli flag for setting all config values for the duration of the command run. I think that for our purposes thats probably very extreme since we expect this to generally be set and forget, or one-off runs for backfill so I'm not suggesting we do this for the indexer...but it does make me wonder if this is something we should explore for the new sui CLI when we get to that.

Sorry for venturing a bit off-topic :)

Reply:
I think we definitely need to think holistically about our "philosophy" for configuring the new CLI. It will be more challenging there than here, and I struggled to come up with clear rules even in this setting.